Wikipedia/Code/ExploreViewController.swift

```diff
@@ -6,12 +6,42 @@ class ExploreViewController: ColumnarCollectionViewController, ExploreCardViewCo
 
     public var presentedContentGroupKey: String?
     public var shouldRestoreScrollPosition = false
+    
+    var startingTheme = "black"
+    let currentDateTime = Date()
+    let userCalendar = Calendar.current
 
+    let requestedComponents: Set<Calendar.Component> = [
+        .year,
+        .month,
+        .day,
+        .hour,
+        .minute,
+        .second
+    ]
+    
     @objc public weak var notificationsCenterPresentationDelegate: NotificationsCenterPresentationDelegate?
     @objc public weak var settingsPresentationDelegate: SettingsPresentationDelegate?
 
     // MARK - UIViewController
     
+    func setAutomaticTheme() {
+          
+        let dateToSetTheme = userCalendar.dateComponents(requestedComponents, from: currentDateTime)
+        if let hour = dateToSetTheme.hour {
+            if (hour >= 6 && hour < 17) {
+                startingTheme = "sepia"
+            } else {
+                startingTheme = "dark"
+            }
+        } else {
+            return
+        }
```

Comment:
The bulk of the above could be pulled into the Date extension, and leave the code confirming the change in theme in this function